src/components/visualisation/pie.tsx

```diff
@@ -0,0 +1,73 @@
+import React, { useEffect, useRef } from "react";
+import Chart from "chart.js/auto";
+import useWindowSize from "../../hooks/useWindow";
+
+const generateRandomColor = () => {
+  const letters = "0123456789ABCDEF";
+  let color = "#";
+  for (let i = 0; i < 6; i++) {
+    color += letters[Math.floor(Math.random() * 16)];
+  }
+  return color;
+};
+
+const PieChart = ({ data }) => {
+  const chartRef = useRef(null);
+  const { height } = useWindowSize();
+
+  useEffect(() => {
+    // Generate dynamic background colors for each segment based on labels
```

Comment:
since it's reading data from props,so there are chances that when you reload or during initiation time data may be undefined which will break the condition data.labels.map and return error.So I suggest you to use optional chaining like data?.labels?.map

Reply:
I have updated it in the recent push.